### PR TITLE
Added tensor-based method to calculate Koopman generator called tgEDMD

### PR DESCRIPTION
**Pull request by lueckem**
This pull requests adds a tensor-based method for calculating the Koopman generator called tgEDMD.
The main functions for this method are in "scikit_tt/data_driven/tgedmd.py".
I also added tests in "tests/tgedmd.py" and an example in "examples/lemon_slice_tgedmd.py".

Because tgEDMD needs to access derivatives of basis functions, I replaced the functions that provide the basis functions in "scikit_tt/data_driven/transform.py" with classes. Previous usage is not impacted by this change.

### CODE REVIEW

File: scikit_tt/data_driven/regression.py [hunk 1]
```diff
@@ -296,7 +297,7 @@ def __arr_construct_stack_left(i, stack_left, x_data, basis_list, solution):
         m = x_data.shape[1]
 
         # contract previous stack element with solution and TDT cores
-        stack_left[i] = np.array([basis_list[i-1][k](x_data) for k in range(n)])
+        stack_left[i] = np.array([[basis_list[i-1][k](x_data[:, j]) for j in range(m)] for k in range(n)])
```
Comment (lueckem) [inline scikit_tt/data_driven/regression.py:300]: Unfortunately you cannot feed the new classes of basis functions data arrays. Only single points.

File: scikit_tt/tensor_train.py [hunk 2]
```diff
@@ -1381,10 +1470,9 @@ def qtt2tt(self, merge_numbers):
 
         return tt_tensor
 
-    def svd(self, index, threshold=0, ortho_l=True, ortho_r=True, overwrite=False):
+    def svd(self, index, threshold=0, max_rank=np.infty, ortho_l=True, ortho_r=True, overwrite=False):
```
Comment (lueckem) [inline scikit_tt/tensor_train.py:1473]: I added the option to cap the maximum ranks of the resulting TT.

File: tests/test_transform.py [hunk 3]
```diff
@@ -1,10 +1,235 @@
 # -*- coding: utf-8 -*-
 
 from unittest import TestCase
+
 import numpy as np
+
 import scikit_tt.data_driven.transform as tdt
 
 
+class TestBasisFunctions(TestCase):
```
Comment (lueckem) [inline tests/test_transform.py:10]: Tests for the new classes of basis functions

**Comment by PGelss** [comment]
Thank you very much for your efforts. That's great work.

**Pull request merged**